### ISSUE TITLE
After update 0.9.0, previous backups are always restored.

### ISSUE THREAD

**Issue by kthugs** [Cleod9/syncmarx-webext#46]
In version 0.9.0, when “Sync Every (Minutes):” time comes, the previous data backed up on Google Drive is always restored.
That is, if "Sync Every (Minutes): 15"
All bookmarks that have been deleted, added, or modified will disappear and be replaced with data backed up to Google Drive every 15 minutes.

**Comment by Cleod9** [comment]
Yikes, I'm surprised i didn't catch that when testing the latest 😬 

I'll take a look at this. What browser are you using by the way?

**Comment by kthugs** [comment]
I'm using chrome
The same thing happened with Edge.

**Comment by sevenjay** [comment]
the same situation,
thank you, I'm also using chrome

**Comment by Cleod9** [comment]
Just a heads up that I did start looking into this, it's strange I don't quite see the same issue. There is definitely a bug in both Firefox and Chromium browsers where upon launch the extension doesn't authenticate properly until you interact with the icon a second time. Definitely some backgrounding issue there. But it doesn't seem to be blowing away changes each sync. I'll keep investigating to see if I can address the issue with the background state of the extension, hopefully fixing that will address the issue. With manifest v3 the extension is able to sleep a lot more easily so it might be causing some issues when it's idle for too long.

**Comment by kthugs** [comment]
Whatever changes are made to my bookmarks, they revert back to the state of the bookmarks backed up to Google Drive after 15 minutes (the default for automatic sync).
My experience is that unless I do a "force push", Google Drive's backups are not updated and I download out-of-date backups every 15 minutes.

"There is definitely a bug in both Firefox and Chromium browsers where upon launch the extension doesn't authenticate properly until you interact with the icon a second time"
I also experienced this.

Currently I'm sideloading 0.8.2
It works perfectly as before.

**Comment by Cleod9** [comment]
Hi all, sorry for the lack of updates.... I did manage to spend a few hours looking into it but to no avail so far. Unfortunately my time is extremely limited at the moment so it will be tough for me to quickly resolve. I heavily suspect this to be related to the manifest v2 to v3 upgrade specifically though, previously in v2 it was easier to keep the tool running as a persistent background process, but now it sort of goes to sleep and spins back up whenever the sync timer goes off. Strange things seem to happen as a result of this (such as the auth issue at launch) and I'm still trying to pinpoint why. Debugging sadly has gotten much much more difficult with this new lifecycle since many errors that occur are lost in the ether

**Comment by beville** [comment]
Just wanted to chime in here that I think I've hit the same root issue, and have been seeing it since mid 2024.

Steps:

1. Start up the browser (Chrome or Firefox), 
2. Click on the pinned extension icon in the toolbar, it shows the set up dialog (asking me to choose a provider, and paste the auth token)
3. Do nothing, just close it by clicking elsewhere
4. Click a second time on the icon, and it shows that it's already authorized, and I can to a sync, push, or pull
5. Wait for a bit, and then back to item 2 above

This happens on Windows or Ubuntu

I've trained myself to be more pro-active about the clicks to force the connect, but still forget sometimes, and loose bookmarks occasionally because of the extra steps.

**Comment by Cleod9** [comment]
Yeah that's definitely the same issue, I'll try my best to take another stab at this soon since it bothers me as well as a user. Wish me luck

**Comment by beville** [comment]
For what it’s worth, I use another extension, “Checker Plus For Gmail” (https://addons.mozilla.org/en-US/firefox/addon/checker-plus-gmail/) that seems to be operating in the background to provide notifications.  I have no solid idea about how these extensions work, so it’s possible it works on a different principle that syncmarx. But I did peek inside the other day and it’s also on Manifest v3.  Maybe there are some hints there?

Good luck! 🤞

**Comment by sigtrap** [comment]
The manifest is missing the permission "background", which may be why it's not starting when chrome loads initially.
https://developer.chrome.com/docs/extensions/reference/permissions-list

**Comment by Cleod9** [comment]
I believe I've identified the issue with the extension not kicking on at startup and should have a fix soon, it has to do with needing an `onStartup` listener in this new service worker ecosystem. I made an assumption that the extension would run its code at least once when the extension is loaded, but apparently that is not the case anymore with V3.

**Comment by beville** [comment]
Nice!  0.9.1 showed up this morning.  Already looks better.  Thanks very much for working on this.

**Comment by Cleod9** [comment]
Great, that's one issue down! i think there is still an issue with the syncing but at least the extension is starting up correctly now. I'll report back when i find out more

**Comment by Cleod9** [comment]
Ok this should now be fixed in 0.9.2, hopefully this gets us back to 0.8.x parity 🤞 

Closing this as completed
